fish_prompt.fish

```diff
@@ -367,6 +368,23 @@ function __bobthefish_prompt_dir -d 'Display a shortened form of the current dir
   __bobthefish_path_segment "$PWD"
 end
 
+function __bobthefish_prompt_vi -d 'Display vi mode'
+  [ "$theme_display_vi" = 'no' ]; and return
+  set -l vi_mode
+  switch $fish_bind_mode
+  case default
+    __bobthefish_start_segment $__bobthefish_med_grey $__bobthefish_dk_grey --bold
+    echo -n -s 'N'
```

Comment:
Default probably shouldn't show anything at all … especially as most users won't use vi mode at all, so adding a segment to their prompt which never changes and adds nothing is kind of a crappy thing to do :)

But even for the users who do, it probably shouldn't show in normal mode. bobthefish goes to a lot of effort to show you only the things which are important. For example, there's no "last exit code was fine" flag, but there is a "last exit code was an error". There's no "currently up to date" git flag, but there is one for each possible permutation of ahead, behind, stashed, dirty, untracked, etc.